### ISSUE TITLE
[v1.1] Model Adapter demo

### ISSUE THREAD

**Issue by XinyaoWa** [intel/e2eAIOK#160]
- [x] Summary
- [x] Built-in demos
    - [x] Finetuner
    - [x] Distiller
    - [x] Domain Adapter
- [x] API usage for customized cases
    - [x] Finetuner
    - [x] Distiller
        - [x] basic
        - [x] save logits
        - [x] train with saved logits
    - [x] Domain Adapter
